pstatus-graphql-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/CustomSchemaGeneratorHooks.kt

```diff
@@ -1,15 +1,135 @@
 package gov.cdc.ocio.processingstatusapi.plugins
 
 import com.expediagroup.graphql.generator.hooks.SchemaGeneratorHooks
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.JsonNodeFactory
+import gov.cdc.ocio.processingstatusapi.collections.BasicHashMap
+import graphql.GraphQLContext
+import graphql.execution.CoercedVariables
+import graphql.language.*
 import graphql.scalars.ExtendedScalars
 import graphql.scalars.datetime.DateTimeScalar
-import graphql.scalars.`object`.JsonScalar
-import graphql.schema.GraphQLScalarType
-import graphql.schema.GraphQLType
+import graphql.schema.*
 import java.time.OffsetDateTime
+import java.util.*
 import kotlin.reflect.KClass
 import kotlin.reflect.KType
 
+
+/**
+ * Converts an [ObjectValue] to a [BasicHashMap].
+ *
+ * @param objectValue ObjectValue
+ * @return BasicHashMap<String, Any?>
+ */
+fun objectValueToHashMap(objectValue: ObjectValue): BasicHashMap<String, Any?> {
+    val jsonNode = objectValueToJsonNode(objectValue)
+    return jsonNodeToHashMap(jsonNode)
+}
+
+/**
+ * Generates a [BasicHashMap] from a provided [JsonNode].
+ *
+ * @param jsonNode JsonNode
+ * @return BasicHashMap<String, Any?>
+ */
+fun jsonNodeToHashMap(jsonNode: JsonNode): BasicHashMap<String, Any?> {
+    val hashMap = BasicHashMap<String, Any?>()
+
+    jsonNode.fields().forEach { (key, value) ->
+        hashMap.put(key, when {
+            value.isObject -> jsonNodeToHashMap(value) // Recursive call for nested objects
+            value.isArray -> value.map { node ->
+                if (node.isObject) jsonNodeToHashMap(node) else node.asText()
+            }
+            value.isDouble -> value.doubleValue()
+            value.isFloat -> value.floatValue()
+            value.isBoolean -> value.asBoolean()
+            value.isBigInteger -> value.asLong()
+            value.isInt -> value.intValue()
+            value.isLong -> value.longValue()
+            value.isNull -> null
+            else -> value.asText()
+        })
+    }
+
+    return hashMap
+}
+
+/**
+ * Converts an [ObjectValue] to a [JsonNode].
+ *
+ * @param objectValue ObjectValue
+ * @return JsonNode
+ */
+fun objectValueToJsonNode(objectValue: ObjectValue): JsonNode {
+    val objectNode = ObjectMapper().createObjectNode()
+
+    objectValue.objectFields.forEach { field ->
+        val key = field.name
+        val value = convertValue(field.value)
+        objectNode.set<JsonNode>(key, value)
+    }
+
+    return objectNode
+}
+
+/**
+ * Converts a [Value] generic to a [JsonNode].
+ *
+ * @param value Value<*>
+ * @return JsonNode
+ */
+private fun convertValue(value: Value<*>): JsonNode {
+    return when (value) {
+        is StringValue -> JsonNodeFactory.instance.textNode(value.value)
+        is IntValue -> JsonNodeFactory.instance.numberNode(value.value)
```

Comment:
Should we check for "double" type as well?

Reply:
We don't need a "DoubleValue" one here, which doesn't exist.  The FloatValue one under the hood is a BigDecimal type, which covers doubles and floats.